GooglePlayInstant/Editor/PlayInstantLoadingScreenGenerator.cs

```diff
@@ -0,0 +1,103 @@
+        /// downloaded from the CDN. Takes in a loadingScreenImagePath, a path to the image shown in the loading scene,
+        /// and an assetbundle URL.
+        /// </summary>
+        public static void GenerateLoadingScreenScene(string loadingScreenImagePath, string assetBundleUrl)
+        {
+            if (!File.Exists(loadingScreenImagePath))
+            {
+                Debug.LogError("Loading screen image file cannot be found.");
+            }
+            else
+            {
+                // Removes the loading scene if it is present
+                EditorSceneManager.CloseScene(SceneManager.GetSceneByName("play-instant-loading-screen-scene"), true);
+
+                var loadingScreenScene = EditorSceneManager.NewScene(NewSceneSetup.EmptyScene, NewSceneMode.Additive);
+                var loadingScreenGameObject = new GameObject("Canvas");
+
+                GenerateLoadingScreenScript(assetBundleUrl);
+                AddLoadingScreenImageToScene(loadingScreenGameObject, loadingScreenImagePath);
+                AddLoadingScreenScript(loadingScreenGameObject);
+
+                EditorSceneManager.SaveScene(loadingScreenScene, "play-instant-loading-screen-scene.unity");
+            }
+        }
+
+        //TODO: get rid of error message associated with using the LoadingScreenScript reference
+        private static void AddLoadingScreenScript(GameObject loadingScreenGameObject)
+        {
+            loadingScreenGameObject.AddComponent<LoadingScreenScript>();
+        }
+
+        private static void AddLoadingScreenImageToScene(GameObject loadingScreenGameObject, string pathToLoadingScreenImage)
+        {
+            loadingScreenGameObject.AddComponent<Canvas>();
+            var loadingScreenCanvas = loadingScreenGameObject.GetComponent<Canvas>();
+            loadingScreenCanvas.renderMode = RenderMode.ScreenSpaceOverlay;
+
+            var loadingScreenImageData = File.ReadAllBytes(pathToLoadingScreenImage);
+            var tex = new Texture2D(1, 1);
+            tex.LoadImage(loadingScreenImageData);
+
+            var loadingImageSprite = Sprite.Create(tex, new Rect(0, 0, tex.width, tex.height), new Vector2(0.5f, 0.5f));
+
+            loadingScreenGameObject.AddComponent<Image>();
+            var loadingScreenImage = loadingScreenGameObject.GetComponent<Image>();
+            loadingScreenImage.sprite = loadingImageSprite;
+        }
+
+        //TODO: add better handling of finding assets folder and figure out possible alternative AssetDatabase synchronous importing
+        private static void GenerateLoadingScreenScript(string assetBundleUrl)
+        {
+            var newloadingScreenScriptDir = "Assets/GooglePlayInstantScript/LoadingScreenScript.cs";
+            var genericLoadingScriptDirStrings = Directory.GetFiles(Directory.GetCurrentDirectory(),
+                "GenericLoadingScreenScript.cs",
```

Comment:
Why do we have to search for this? Shouldn't we already know the path?

Reply:
What if the developer changes the path of the plugin, but it's still there? (e.g. in a folder). Similarly for the line above, what if the Assets folder is in another folder, and not necessarily in a top level directory for the project?